### ISSUE TITLE
Working well for batch size <=32 for single gpu and multi gpu training. While getting following error when using batch size > 32 for multi gpu. Plz help me to fix this.

### ISSUE THREAD

**Issue by sanjaydeo96** [horovod/horovod#2170]
**Environment:**
1. Framework: (TensorFlow 1.5)
2. Framework version:
3. Horovod version: Not using horovod instead running through openmpi command
4. MPI version: 4.04
5. CUDA version: 11
6. NCCL version: 2
7. Python version: 3.6
9. OS and version: 18.04

    for grad in grads]
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/horovod/tensorflow/__init__.py", line 257, in <listcomp>
    for grad in grads]
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/horovod/tensorflow/__init__.py", line 105, in allreduce
    summed_tensor_compressed = _allreduce(tensor_compressed, op=true_op)
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/horovod/tensorflow/mpi_ops.py", line 107, in _allreduce
    return MPI_LIB.horovod_allreduce(tensor, name=name, reduce_op=op)
  File "<string>", line 84, in horovod_allreduce
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/tensorflow_core/python/framework/op_def_library.py", line 794, in _apply_op_helper
    op_def=op_def)
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/tensorflow_core/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/tensorflow_core/python/framework/ops.py", line 3357, in create_op
    attrs, op_def, compute_device)
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/tensorflow_core/python/framework/ops.py", line 3426, in _create_op_internal
    op_def=op_def)
  File "/home/ubuntu/anaconda3/envs/eff/lib/python3.6/site-packages/tensorflow_core/python/framework/ops.py", line 1748, in __init__
    self._traceback = tf_stack.extract_stack()

--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[41109,1],4]
  Exit code:    1
--------------------------------------------------------------------------

**Comment by tgaddair** [comment]
Hey @sanjaydeo96, looks like the top of the stack trace is missing.  Can you share the full log output?

**Comment by sanjaydeo96** [comment]
So I found the bug. The issue was I was using horovod in which batch size is automatically adjusted with #gpu so I was getting this when I was using a batch size which is greater than full capacity for single gpu.
I have one question btw.
**I am getting this Zigzag curve in horovod multi gpu training.I think issue may be due to optimizer and learning rate. Do you have any suggestion @tgaddair ?**
![Screenshot from 2020-08-11 11-26-17](https://user-images.githubusercontent.com/29541116/89862690-1d4fd480-dbc6-11ea-8279-8c3828403c0d.png)

**Comment by tgaddair** [comment]
Hey @sanjaydeo96, when using Horovod you should keep the batch size per worker the same, but scale the learning rate.  That should help with controlling the loss.  Also, make sure you are only logging metrics on rank 0 (`if hvd.rank() ==0`), as that may be causing some of the tensorboard issues you're seeing.
